src/Plugin/facets/query_type/LocalGovDirectoriesQueryType.php

```diff
@@ -63,11 +64,48 @@ public function execute() {
    * {@inheritdoc}
    */
   public function build() {
+    $results = $this->results;
     $query_operator = $this->facet->getQueryOperator();
+    $field_identifier = $this->facet->getFieldIdentifier();
 
-    if (!empty($this->results)) {
+    // Get passed in facets.
+    $temp_query = $this->query->getOriginalQuery();
+    $temp_query->preExecute();
+    $conditions = &$temp_query->getConditionGroup()->getConditions();
+
+    // Find all the facet condition groups.
+    $facet_conditions = [];
+    foreach ($conditions as $key => $condition) {
+      if ($condition instanceof ConditionGroupInterface) {
+        $tags = $condition->getTags();
+        foreach($tags as $tag) {
+          if (strpos($tag, 'facet:' . $field_identifier . '.') === 0) {
+            $facet_conditions[$key] = $tag;
+          }
+        }
+      }
+    }
+
+    // Run query elimnating each facet group and return the resulting facets.
+    $results = $this->results;
+    foreach($facet_conditions as $key => $filter_tag) {
+      $group_facets = $this->getResultingFacetsFromFacetGroupExceptOwn($filter_tag);
+      
+      // Remove any duplicate facets, or they show up multiple times.
+      $group_facets_filtered = array_filter($group_facets, function($item) use ($results) {
+        $facet_ids = array_column($results, 'filter');
+        if (in_array($item['filter'], $facet_ids)) {
+          return FALSE;
+        }
+        return TRUE;
+      });
+      $results = array_merge($results, $group_facets_filtered);
```

Comment:
Theres still a risk that the active chosen facets get filtered, and theres no way of unticking them, so we need a way to keep in the existing facets.